_posts/2023-10-03-securing-wildfly-apps-oidc-openshift.adoc

```diff
@@ -0,0 +1,360 @@
+
+* Examine the https://github.com/wildfly-security-incubator/elytron-examples/blob/main/simple-webapp-oidc/src/main/webapp/WEB-INF/oidc.json[oidc.json] file. The `oidc.json` is used to configure the native OIDC client subsystem.
++
+[source,options="nowrap"]
+----
+{
+    "client-id" : "myclient",                                                         <1>
+    "provider-url" : "${env.OIDC_PROVIDER_URL:http://localhost:8080}/realms/myrealm", <2>
+    "public-client" : "true",                                                         <3>
+    "principal-attribute" : "preferred_username",                                     <4>
+    "ssl-required" : "EXTERNAL"                                                       <5>
+}
+----
+<1> This is the client we created in Keycloak.
+<2> The provider URL, which is the URL for the realm *myrealm* that we created, is specified as an environment variable. We will set its value in the helm configuration.
+<3> When *public-client* set to *true*, client credentials are not sent when communicating with the OpenID provider.
+<4> We specify that the user name of the identity, which in our case is *alice*, is to be used as the principal for the identity.
+<5> When *ssl-required* is set to *EXTERNAL*, only the communication with external clients happens over HTTPs
+
+== Get the Application URL
+
+Once the WildFly server has been provisioned, use the following command to find the URL for your example
+application:
+
+[source]
+----
+SIMPLE_WEBAPP_OIDC_URL=https://$(oc get route oidc-app --template='{{ .spec.host }}') &&
+echo "" &&
+echo "Application URL: $SIMPLE_WEBAPP_OIDC_URL/simple-webapp-oidc"  &&
+echo "Valid redirect URI: $SIMPLE_WEBAPP_OIDC_URL/simple-webapp-oidc/secured/*" &&
+echo ""
+----
+
+We'll make use of these URLs in the next two sections.
+
+== Finish Configuring Keycloak
+
+From your *myclient* client in the Keycloak Administration Console,
+in the client settings, set *Valid Redirect URI* to the Valid redirect URI that was output in the previous section and then click *Save*.
+
+== Access the Application
+
+From your browser, navigate to the *Application URL* that was output in the previous section.
+
+Click on *Access Secured Servlet*.
+
+You will be redirected to Keycloak to log in.
+
+Log in using the *alice* user we created earlier.
+
+Upon successful authentication, you will be redirected back to the example application.
+
+The example application simply outputs the name of the logged in user.
+
+You should see the following output:
+
+```
+Secured Servlet
+
+Current Principal 'alice'
+```
+
+This indicates that we have successfully logged into our application!
+
+== Summary
+
+This guide has shown how to secure an application deployed to WildFly on OpenShift with OIDC. For additional
+information, feel free to check out the resources linked below.
+
+== Resources
+
+* https://docs.wildfly.org/29/Getting_Started_on_OpenShift.html[Getting Started with WildFly on OpenShift]
```

Comment:
We should update the WildFly docs URLs to reference 30 instead of 29.

Reply:
fixed